Build error using external/system fmt library

The C++ fmt library is incorrectly configured to always use the include path in the submodule, even when linking against out-of-core fmt.

```bash
cmake .. -DARB_USE_BUNDLED_LIBS=off -G Ninja
ninja modcc
[13/25] Building CXX object modcc/CMakeFiles/libmodcc.dir/printer/cprinter.cpp.o
FAILED: modcc/CMakeFiles/libmodcc.dir/printer/cprinter.cpp.o 
/usr/bin/c++  -I/home/bcumming/software/github/arbor-fmt/modcc/. -I/home/bcumming/software/github/arbor-fmt/modcc/../ext/fmt/include -O3 -DNDEBUG -Wall -Wno-maybe-uninitialized -Wno-psabi -std=c++17 -MD -MT modcc/CMakeFiles/libmodcc.dir/printer/cprinter.cpp.o -MF modcc/CMakeFiles/libmodcc.dir/printer/cprinter.cpp.o.d -o modcc/CMakeFiles/libmodcc.dir/printer/cprinter.cpp.o -c /home/bcumming/software/github/arbor-fmt/modcc/printer/cprinter.cpp
/home/bcumming/software/github/arbor-fmt/modcc/printer/cprinter.cpp:17:10: fatal error: fmt/core.h: No such file or directory
   17 | #include <fmt/core.h>
      |          ^~~~~~~~~~~~
compilation terminated.
```
Note the `-I/home/bcumming/software/github/arbor-fmt/modcc/../ext/fmt/include`

--- a/modcc/CMakeLists.txt
+++ b/modcc/CMakeLists.txt
@@ -31,19 +31,18 @@ set(modcc_sources modcc.cpp)
 
 add_library(libmodcc STATIC ${libmodcc_sources})
 target_include_directories(libmodcc PUBLIC .)
-target_include_directories(libmodcc PRIVATE ../ext/fmt/include)
+if (ARB_USE_BUNDLED_FMT)
+    target_include_directories(libmodcc PRIVATE ../ext/fmt/include)
+    target_compile_definitions(libmodcc PRIVATE FMT_HEADER_ONLY)
+else ()
+    find_package(fmt REQUIRED)
+    target_link_libraries(libmodcc PRIVATE fmt::fmt-header-only)
+endif ()
 
 set_target_properties(libmodcc PROPERTIES OUTPUT_NAME modcc)
 
-
 add_executable(modcc ${modcc_sources})
 target_link_libraries(modcc PRIVATE libmodcc ext-tinyopt)
-if (ARB_USE_BUNDLED_FMT)
-    target_include_directories(modcc PRIVATE ../ext/fmt/include)
-else ()
-    find_package(fmt)
-    target_link_libraries(modcc PRIVATE fmt::fmt-header-only)
-endif ()
 set_target_properties(modcc libmodcc PROPERTIES EXCLUDE_FROM_ALL ${ARB_WITH_EXTERNAL_MODCC})
 
 if (NOT ARB_WITH_EXTERNAL_MODCC)

--- a/modcc/printer/cprinter.cpp
+++ b/modcc/printer/cprinter.cpp
@@ -13,7 +13,6 @@
 #include "printer/printerutil.hpp"
 #include "printer/marks.hpp"
 
-#define FMT_HEADER_ONLY YES
 #include <fmt/core.h>
 #include <fmt/format.h>
 #include <fmt/compile.h>

--- a/modcc/printer/gpuprinter.cpp
+++ b/modcc/printer/gpuprinter.cpp
@@ -4,7 +4,6 @@
 #include <set>
 #include <regex>
 
-#define FMT_HEADER_ONLY YES
 #include <fmt/core.h>
 #include <fmt/format.h>
 #include <fmt/compile.h>

--- a/modcc/printer/infoprinter.cpp
+++ b/modcc/printer/infoprinter.cpp
@@ -3,7 +3,6 @@
 #include <string>
 #include <regex>
 
-#define FMT_HEADER_ONLY YES
 #include <fmt/core.h>
 #include <fmt/format.h>
 #include <fmt/compile.h>

